packages/beacon-state-transition/src/allForks/util/cachedBeaconState.ts

```diff
@@ -120,6 +129,8 @@ export class BeaconStateContext<T extends allForks.BeaconState> {
     balanceCache: MutableVector<T["balances"][number]>,
     previousEpochParticipationCache: MutableVector<IParticipationStatus>,
     currentEpochParticipationCache: MutableVector<IParticipationStatus>,
+    currentSyncCommittee: IndexedSyncCommittee | undefined,
+    nextSyncCommittee: IndexedSyncCommittee | undefined,
```

Comment:
Should we not make this properties optional and initialize them to empty on phase0? Would that be very wasteful?

Reply:
it's not wasteful but why do we want to do that? If `undefined` is not clear enough, we can make it `null`

Reply:
Not having optional types is great for consumer code. We did a big refactor a few months ago around the strictNullChecks to not have optional state when avoidable.

Reply:
👍 will do that, we have the same thing for `cachedPreviousParticipation`